kernel/include/paging.h

```diff
@@ -9,12 +9,19 @@
 #include <stdint.h>
 #include <stdbool.h>
 
+typedef uint64_t PageFlags;
+typedef uint64_t PageEntry;
+typedef uint64_t* PageTable;
```

Comment:
It might make more sense to not have the PageTable typedef and instead just return a pointer to a PageEntry. It would be more clear that you are returning a pointer instead of a value.

Reply:
This would create ambiguity between PageTables and pointers to a PageEntry (ie recieved from fetch_page), which i believe should be treated as different objects in code.

Reply:
But aren't page tables literally an array of page entries? I think that masking the fact that you are returning a pointer is much worse.